frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -123,40 +128,20 @@ impl syn::parse::Parse for DisableFrameSystemSupertraitCheck {
 	}
 }
 
-/// Parse for `#[pallet::constant]`
-pub struct TypeAttrConst {
-	pound_token: syn::Token![#],
-	bracket_token: syn::token::Bracket,
-	pallet_ident: syn::Ident,
-	path_sep_token: syn::token::PathSep,
-	constant_keyword: keyword::constant,
+/// Parsing for `#[pallet::X]`
+#[derive(derive_syn_parse::Parse)]
+pub struct PalletAttr {
+	_pound: Token![#],
+	#[bracket]
+	_bracket: token::Bracket,
+	#[inside(_bracket)]
+	_pallet: keyword::pallet,
+	#[inside(_bracket)]
+	_sep: Token![::],
+	#[inside(_bracket)]
+	ident: Ident,
```

Comment:
Instead of permitting an `Ident` here, I propose that we bring back `constant` and `no_default` as keywords, and then we create a new enum that encapsulates the two and put that new enum as the type here.